test/clj/rems/test_browser.clj

```diff
@@ -159,12 +168,17 @@
 (defn fill-form-field
   "Fills a form field named by `label` with `text`.
 
-  Optionally give `:index` when several items match. It starts from 1."
+  Optionally give `:index` when several items match. It starts from 1.
+  Optionally give `:form-nth` when several forms exist. It starts from 1."
   [label text & [opts]]
   (assert (> (:index opts 1) 0) "indexing starts at 1") ; xpath uses 1, let's keep the convention though we don't use xpath here because it will likely not work
+  (assert (> (:form-nth opts 1) 0) "indexing starts at 1") ; css pseudo-selector works the same way as above
 
-  (let [el (nth (btu/query-all [{:css ".fields"}
-                                {:tag :label :fn/has-text label}])
+  (let [el (nth (btu/query-all (into (if-let [form-nth (:form-nth opts)]
+                                       [{:css (str ".application-fields:nth-child(" form-nth ")")}]
+                                       [])
+                                     [{:css ".fields"}
+                                      {:tag :label :fn/has-text label}]))
```

Comment:
this should stay compatible with other tests that utilize `fill-form-field` function. `:index` refers to multiple form fields (i believe), while `:form-nth` refers to multiple `div.application-fields` elements. the separation is necessary because some views contain only `div.fields`, and so the `.application-fields:...` would not target those fields at all.

Reply:
css `nth-child` selector targets siblings. if you have multiple forms, they are under separate `.application-fields` elements which makes them not siblings. i tried passing xpath `:index` selector in but i could not get it working with `.fields`. so in other words this is a workaround solution

Reply:
How about you `query-all` `.fields` and then just take the `nth` of that in Clojure? And then inside that element query for the label with the possible field index?

Reply:
`etaoin` documentation leaves quite a lot to desire, but i managed to find the `children` wrapping function that allows just this. this should work better :)